src/Microsoft.DotNet.Darc/Darc/Options/CommandLineOptions.cs

```diff
@@ -178,7 +171,7 @@ public virtual IServiceCollection RegisterServices(IServiceCollection services)
             return new VmrInfo(string.Empty, string.Empty);
         });
         services.TryAddSingleton<IRedisCacheClient, NoOpRedisClient>();
-        services.TryAddSingleton<ICommentCollector, CommentCollector>();
+        services.TryAddScoped<ICommentCollector, CommentCollector>();
```

Comment:
Does this make sense in darc?

Reply:
It does because comment collector is used in some classes to print into console and if we have commands not using codeflow, they still might need a dependency which relies on this.

Basically, I tried it and some darc commands were failing.